js/mirador_viewer.js

```diff
@@ -14,10 +14,57 @@
         attach: function (context, settings) {
             Object.entries(settings.mirador.viewers).forEach(entry => {
               const [base, values] = entry;
-              once('mirador-viewer', base, context).forEach(() =>
+              once('mirador-viewer', base, context, settings).forEach(() => {
+                if (settings.token !== undefined) {
+                  values["resourceHeaders"] = {
+                    'Authorization': 'Bearer '+ settings.token,
+                    'token': settings.token
+                  };
+                  values["requestPipeline"] = [
+                    (url, options) => ({  ...options, headers: {
+                      "Accept": 'application/ld+json;profile="http://iiif.io/api/presentation/3/context.json"',
+                      'Authorization': 'Bearer '+ settings.token,
+                      'token': settings.token
+                    }})
+                  ];
+                  values["osdConfig"] = {
+                    "loadTilesWithAjax": true,
+                    "ajaxHeaders": {
+                      'Authorization': 'Bearer '+ settings.token,
+                      'token': settings.token
+                    }
+                  };
+                  values["requests"] = {
+                    preprocessors: [ // Functions that receive HTTP requests and manipulate them (e.g. to add headers)
+                      // rewrite all info.json requests to add the text/json request header
+                      (url, options) => (url.match('info.json') && { ...options, headers: {
+                        'Authorization': 'Bearer '+ settings.token,
+                        'token': settings.token
+                      }})
+                    ],
+                };
+
+                }
                 Mirador.viewer(values, window.miradorPlugins || {})
+                }
               );
             });
+            if (settings.token !== undefined) {
+              if ('serviceWorker' in navigator) {
+                // The Mirador viewer uses img tags for thumbnails so thumbnail image requests
+                // do not have authorization or token headers. Attach them using a service worker.
+                window.addEventListener('load', () => {
+                  navigator.serviceWorker
+                    .register('/islandora_mirador_service_worker?token=' + settings.token, { scope: '/' })
```

Comment:
Not sure this is fully portable, with the assumption of the site being directly on the root? As in, if using language or site path prefixing, then should this follow suit? Drupal might expose a helper for this?